azext_iot/operations/events3/_events.py

```diff
@@ -71,45 +71,104 @@ def stop_and_suppress_eloop():
                 raise RuntimeError(error)
 
 
-async def initiate_event_monitor(target, consumer_group, enqueued_time, device_id=None,
-                                 properties=None, timeout=0, output=None, content_type=None,
-                                 devices=None, interface_id=None, pnp_context=None):
-    def _get_conn_props():
-        properties = {}
-        properties["product"] = "az.cli.iot.extension"
-        properties["version"] = VERSION
-        properties["framework"] = "Python {}.{}.{}".format(*sys.version_info[0:3])
-        properties["platform"] = sys.platform
-        return properties
+def buildCentralEventHubTargetSync(cmd, app_id, aad_token):
```

Comment:
There is a big opportunity for refactoring here. Please take all these 'buildXTarget...'  functions and instead refactor them into a class - something like `EventTargetBuilder` (open to names)  - the EventTargetBuilder should be able to build central or iothub targets. Try and keep as much of the redundancy out. Introducing this class (the goal anyway) should clean-up `_events.py` flow quite a bit.

Reply:
Thanks for the tips, I'll refactor